### ISSUE TITLE
Update vector3's XYZ to use auto property instead of backing field

### ISSUE THREAD

**Issue by tekinomikata** [Eastrall/Rhisis#302]
I really don't remember why I've done that at the begining of the project. Maybe we could replace the X, Y and Z properties defintion with `{ get; set; }` instead of having a private variable ?

_Originally posted by @Eastrall in https://github.com/Eastrall/Rhisis/pull/301_

**Comment by tekinomikata** [comment]
Will do that, I don't see any reason not to, or couldn't think of any unless it was a struct

**Comment by tekinomikata** [comment]
@Eastrall there is just 1 issue with making Vector3 auto prop, not elegant but can be resolved with:
```
        public Vector3(string x, string y, string z)
        {
            float.TryParse(x, out var result);
            X = result;
            float.TryParse(y, out result);
            Y = result;
            float.TryParse(z, out result);
            Z = result;
        }
```
Since you can't use X directly in the out field. 

And the GetHashCode because Vector3 allows for modification

**Comment by Eastrall** [comment]
I think the second solution is better. We need to be careful with the formats thought, because if the format is incorrect and the method can not parse the value, it will throw a `FormatException`. But maybe that what we need to do in order to catch errors.

**Comment by tekinomikata** [comment]
@Eastrall that method is only used to parse the data into LUA right?

So I should leave it for whatever code that is using to catch it or you want me to catch it there and log?

I suppose you would want to let it throw there in that method and catch where you will be calling it?

**Comment by Eastrall** [comment]
It's used to load the map and region information only during startup time. I think we can use the `Convert.ToSingle()` method.

**Comment by Eastrall** [comment]
Solved in #304
